### ISSUE TITLE
Opendap "Unknown file format"; --has-dap -> yes

### ISSUE THREAD

**Issue by evanmason** [Unidata/netcdf4-python#581]
I have netCDF4 v1.2.4 installed on Linux.  When I try to open a file with Opendap I get:
```
In [3]: url = 'http://dtvirt5.deltares.nl:8080/thredds/dodsC/opendap/rijkswaterstaat/jarkus/profiles/transect.nc'

In [4]: from netCDF4 import Dataset

In [5]: nc = Dataset(url)
---------------------------------------------------------------------------
IOError                                   Traceback (most recent call last)
<ipython-input-5-61cdc91c2328> in <module>()
----> 1 nc = Dataset(url)

netCDF4/_netCDF4.pyx in netCDF4._netCDF4.Dataset.__init__ (netCDF4/_netCDF4.c:12621)()

IOError: NetCDF: Unknown file format
```
This error can happen if the netcdf C lib is not built with support for Opendap ([https://github.com/Unidata/netcdf4-python/issues/509](url)) but in my case I have that support:
```
[emason@marula Downloads]$ nc-config --has-dap
yes
```
and 
```
[emason@marula Downloads]$ ncdump -h http://dtvirt5.deltares.nl:8080/thredds/dodsC/opendap/rijkswaterstaat/jarkus/profiles/transect.nc
netcdf transect {
dimensions:
        time = UNLIMITED ; // (51 currently)
        alongshore = 2268 ;
...
```
I am using netcdf library version 4.4.0.

I have tried installing netCDF4 both at the command line and with pip following the instructions at [http://unidata.github.io/netcdf4-python/](url), and also within a virtual environment, without success.

I'd be grateful for any suggestions, thanks.

**Comment by jswhit** [comment]
Here's what I get:

```python
[mac28:~/Downloads] jwhitaker% python
Python 2.7.11 (default, Mar  1 2016, 18:40:10)
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from netCDF4 import Dataset
>>> url = 'http://dtvirt5.deltares.nl:8080/thredds/dodsC/opendap/rijkswaterstaat/jarkus/profiles/transect.nc'
>>> nc = Dataset(url)
>>> nc
<type 'netCDF4._netCDF4.Dataset'>
root group (NETCDF3_CLASSIC data model, file format DAP2):
    naming_authority: deltares.nl
    id: JarKus_release20150930_origins135
    Metadata_Conventions: Unidata Dataset Discovery v1.0
    title: JarKus Data (cross-shore transects)
    summary: Cross-shore yearly transect bathymetry measurements along the Dutch coast since 1965
    keywords: Bathymetry, JarKus, Dutch coast
    keywords_vocabulary: http://www.eionet.europa.eu/gemet
    standard_name_vocabulary: http://cf-pcmdi.llnl.gov/documents/cf-standard-names/
    history: 2015-09-30T07:54Z: NetCDF created by santinel on win64 computer DIRECTORYV-VML0218 with Matlab version 8.4.0.150421 (R2014b) and main function https://svn.oss.deltares.nl/repos/openearthrawdata/trunk/rijkswaterstaat/jarkus/scripts/jarkus_convert.m (revision 8604)
subversion repositories:
https://svn.oss.deltares.nl/repos/openearthrawdata/trunk/rijkswaterstaat/jarkus (revision 8604P)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab (revision 12211:12212)
primary related functions:
https://svn.oss.deltares.nl/repos/openearthrawdata/trunk/rijkswaterstaat/jarkus/scripts/jarkus_transect.m (revision 8604)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/applications/Rijkswaterstaat/jarkus/jarkus_createtransectstruct.m (revision 12212)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/applications/Rijkswaterstaat/jarkus/jarkus_grid2netcdf.m (revision 12212)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/applications/Rijkswaterstaat/jarkus/jarkus_svnrev_transects.m (revision 12212)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/applications/Rijkswaterstaat/jarkus/jarkus_transect2grid.m (revision 12212)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/applications/Rijkswaterstaat/jarkus/jarkus_transect2netcdf.m (revision 12212)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/applications/Rijkswaterstaat/jarkus/jarkus_updategrid.m (revision 12212)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/dir2.m (revision 12211)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/general/SVN_fun/svn_info.m (revision 12211)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/general/io_fun/abspath.m (revision 12211)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/general/io_fun/getCalls.m (revision 12211)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/general/string_fun/var2evalstr.m (revision 12211)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/general/time_fun/nowutc.m (revision 12211)
https://svn.oss.deltares.nl/repos/openearthtools/trunk/matlab/setproperty.m (revision 12211)
2015-08-31T12:54Z: Raw data received from Rijkswaterstaat
HEADurl: https://svn.oss.deltares.nl/repos/openearthrawdata/trunk/rijkswaterstaat/jarkus/raw (rev 8604P)
Files:
/total/Ameland.txt (rev 5880)
/total/Delfland.txt (rev 5880)
/total/Goeree.txt (rev 5880)
/total/Maasvlakte.txt (rev 5881)
/total/Noord-Beveland.txt (rev 5880)
/total/Noord-Holland.txt (rev 5880)
/total/Rijnland.txt (rev 5880)
/total/Schiermonnikoog.txt (rev 5880)
/total/Schouwen.txt (rev 5880)
/total/Terschelling.txt (rev 5880)
/total/Texel.txt (rev 5880)
/total/Vlieland.txt (rev 5880)
/total/Voorne.txt (rev 5880)
/total/Walcheren.txt (rev 5880)
/total/Zeeuws-Vlaanderen.txt (rev 5880)
/annual/2008/10_msvl08kop.jrk (rev 5876)
/annual/2009/02-schr2009.jrk (rev 5875)
/annual/2009/03-amld2009.jrk (rev 8594)
/annual/2009/04-ters2009.jrk (rev 8594)
/annual/2009/05-vlie2009.jrk (rev 8594)
/annual/2009/06-texl2009.jrk (rev 8594)
/annual/2009/07-nhol2009.jrk (rev 8594)
/annual/2009/08-rijnld-NH2009.jrk (rev 8594)
/annual/2009/08-rijnld-ZH2009.jrk (rev 8594)
/annual/2009/09-delf2009.jrk (rev 8594)
/annual/2009/10-msvl2009.jrk (rev 8594)
/annual/2009/11-vrne2009.jrk (rev 8594)
/annual/2009/12-goer2009.jrk (rev 8594)
/annual/2009/13-schw2009.jrk (rev 8594)
/annual/2009/14-osch2009.jrk (rev 8594)
/annual/2009/15-nbev2009.jrk (rev 8594)
/annual/2009/16-walc2009.jrk (rev 8594)
/annual/2009/17-zwvl2009.jrk (rev 8594)
/annual/2010/02-schr2010.jrk (rev 5879)
/annual/2010/03-amld2010.jrk (rev 1455)
/annual/2010/04-ters2010.jrk (rev 1455)
/annual/2010/05-vlie2010.jrk (rev 1455)
/annual/2010/06-texl2010.jrk (rev 1455)
/annual/2010/07-nhol2010.jrk (rev 1455)
/annual/2010/08-rnld2010.jrk (rev 8597)
/annual/2010/09-delf2010.jrk (rev 1455)
/annual/2010/10-msvl2010.jrk (rev 8597)
/annual/2010/11-vrne2010.jrk (rev 8597)
/annual/2010/12-goer2010.jrk (rev 5898)
/annual/2010/13-schw2010.jrk (rev 1455)
/annual/2010/14-osch10.jrk (rev 8597)
/annual/2010/14-osch2010.jrk (rev 1455)
/annual/2010/15-nbev2010.jrk (rev 1455)
/annual/2010/16-walc2010.jrk (rev 1455)
/annual/2010/17-zwvl2010.jrk (rev 1455)
/annual/2011/02-schr11kop.jrk (rev 5088)
/annual/2011/03-amel11kop.jrk (rev 5088)
/annual/2011/04-ters11kop.jrk (rev 5088)
/annual/2011/05-vlie11kop.jrk (rev 5088)
/annual/2011/06-texl11kop.jrk (rev 5088)
/annual/2011/07-nhol11kop.jrk (rev 5088)
/annual/2011/08-rnld11kop.jrk (rev 5088)
/annual/2011/09-delf11kop.jrk (rev 5088)
/annual/2011/10-msvl11kop.jrk (rev 5878)
/annual/2011/11-vrne11kop.jrk (rev 5088)
/annual/2011/11-vrne11kop2.jrk (rev 5878)
/annual/2011/12-goer11kop.jrk (rev 5898)
/annual/2011/13-schw11kop.jrk (rev 5088)
/annual/2011/14-osch11kop.jrk (rev 5878)
/annual/2011/15-nbev11kop.jrk (rev 5088)
/annual/2011/16-walc11kop.jrk (rev 5088)
/annual/2011/17-zwvl11kop.jrk (rev 5088)
/annual/2012/02-schr12kop.jrk (rev 5428)
/annual/2012/03-amld12kop.jrk (rev 5428)
/annual/2012/04-ters12kop.jrk (rev 5428)
/annual/2012/05-vlie12kop.jrk (rev 5428)
/annual/2012/06-texl12kop.jrk (rev 5428)
/annual/2012/07-nhol12kop.jrk (rev 5428)
/annual/2012/08-rnld12kop.jrk (rev 5428)
/annual/2012/09-delf12kop.jrk (rev 5428)
/annual/2012/10-msvl12kop.jrk (rev 5428)
/annual/2012/11-vrne12kop.jrk (rev 5428)
/annual/2012/12-goer12kop.jrk (rev 5898)
/annual/2012/13-schw12kop.jrk (rev 5877)
/annual/2012/14-osch12kop.jrk (rev 5428)
/annual/2012/15-nbev12kop.jrk (rev 5428)
/annual/2012/16-walc12kop.jrk (rev 5428)
/annual/2012/17-zwvl12kop.jrk (rev 5428)
/annual/2013/02-schr13kop.jrk (rev 5642)
/annual/2013/03-amld13kop.jrk (rev 5642)
/annual/2013/04-ters13kop.jrk (rev 5642)
/annual/2013/05-vlie13kop.jrk (rev 5642)
/annual/2013/06-texl13kop.jrk (rev 5642)
/annual/2013/07-nhol13kop.jrk (rev 5642)
/annual/2013/08-rnld13kop.jrk (rev 5642)
/annual/2013/09-delf13kop.jrk (rev 5703)
/annual/2013/10-msvl13kop.jrk (rev 5642)
/annual/2013/11-vrne13kop.jrk (rev 5642)
/annual/2013/12-goer13kop.jrk (rev 5898)
/annual/2013/13-schw13kop.jrk (rev 5873)
/annual/2013/14-osch13kop.jrk (rev 5642)
/annual/2013/15-nbev13kop.jrk (rev 5642)
/annual/2013/16-walc13kop.jrk (rev 5873)
/annual/2013/17-zwvl13kop.jrk (rev 5642)
/annual/2014/02-schr14kop.jrk (rev 5894)
/annual/2014/03-amld14kop.jrk (rev 5894)
/annual/2014/04-ters14kop.jrk (rev 5894)
/annual/2014/05-vlie14kop.jrk (rev 5894)
/annual/2014/06-texl14kop.jrk (rev 5894)
/annual/2014/07-nhol14kop.jrk (rev 5874)
/annual/2014/08-rnld14kop.jrk (rev 5894)
/annual/2014/09-delf14kop.jrk (rev 5874)
/annual/2014/11-vrne14kop.jrk (rev 5894)
/annual/2014/12-goer14kop.jrk (rev 5898)
/annual/2014/13-schw14kop.jrk (rev 5894)
/annual/2014/14-osch14kop.jrk (rev 5894)
/annual/2014/15-nbev14kop.jrk (rev 5894)
/annual/2014/16-walc14kop.jrk (rev 5894)
/annual/2014/17-zwvl14kop.jrk (rev 5874)
/annual/2015/02-schr15kop.jrk (rev 8593)
/annual/2015/03-amld15kop.jrk (rev 8593)
/annual/2015/04-ters15kop.jrk (rev 8593)
/annual/2015/05-vlie15kop.jrk (rev 8593)
/annual/2015/06-texl15kop.jrk (rev 8593)
/annual/2015/07-nhol15kop.jrk (rev 8593)
/annual/2015/08-rnld15kop.jrk (rev 8593)
/annual/2015/09-delf15kop.jrk (rev 8593)
/annual/2015/10-msvl15kop.jrk (rev 8593)
/annual/2015/11-vrne15-kop.jrk (rev 8593)
/annual/2015/12-goer15kop.jrk (rev 8599)
/annual/2015/13-schw15kop.jrk (rev 8593)
/annual/2015/14-osch15kop.jrk (rev 8593)
/annual/2015/15-nbev15-kop.jrk (rev 8593)
/annual/2015/16-walc15kop.jrk (rev 8593)
/annual/2015/17-zwvl15kop.jrk (rev 8593)
    comment: The transects in this file are a combination of origins: 1: beach_only  3: interpolation  5: sea_only
Due to repositioning of a small number of transects in the period between 1965 and 1970, the position of the following transects can be incorrect in the first years of the measured period: 11001000 11001001 11001003 11001004 11001520 11001540 11001560 11001580 11001600 11001620 11001640 11001660 11001680 11001700 12001900 12001925 12001950 12001975 12002000
Years covered: 1965-2015 (None missing)
    institution: Rijkswaterstaat
    source: on shore and off shore measurements
    references: Original source: http://www.watermarkt.nl/kustenzeebodem/
    Conventions: CF-1.6
    creator_name: Rijkswaterstaat
    creator_url: http://www.rijkswaterstaat.nl
    creator_email: info@rijkswaterstaat.nl
    date_created: 2015-09-30T07:57Z
    date_modified: 2015-09-30T08:08Z
    date_issued: 2015-09-30T08:08Z
    publisher_name: Giorgio Santinelli
    publisher_url: http://www.deltares.nl
    publisher_email: Giorgio.Santinelli@deltares.nl
    geospatial_vertical_units: m
    geospatial_vertical_resolution: 0.01
    geospatial_vertical_positive: up
    processing_level: preliminary
    license: These data can be used freely for research purposes provided that the following source is acknowledged: RIJKSWATERSTAAT. disclaimer: This data is made available in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
    cdm_data_type: grid
    geospatial_vertical_min: -60.85
    geospatial_vertical_max: 49.02
    time_coverage_start: 1965-01-01T00:00Z
    time_coverage_end: 2015-12-31T23:59Z
    time_coverage_resolution: year
    DODS.strlen: 100
    DODS.dimName: stringsize
    DODS_EXTRA.Unlimited_Dimension: time
    dimensions(sizes): time(51), alongshore(2268), bounds2(2), cross_shore(1925), maxStrlen64(64)
    variables(dimensions): int32 id(alongshore), int32 areacode(alongshore), |S1 areaname(alongshore,maxStrlen64), float64 alongshore(alongshore), float64 cross_shore(cross_shore), float64 time(time), float64 time_bounds(time,bounds2), int32 epsg(), float64 angle(alongshore), float64 mean_high_water(alongshore), float64 mean_low_water(alongshore), float64 rsp_x(alongshore), float64 rsp_y(alongshore), float64 rsp_lat(alongshore), float64 rsp_lon(alongshore), float64 x(alongshore,cross_shore), float64 y(alongshore,cross_shore), float64 lat(alongshore,cross_shore), float64 lon(alongshore,cross_shore), int32 max_cross_shore_measurement(time,alongshore), int32 min_cross_shore_measurement(time,alongshore), int32 nsources(time,alongshore), float64 max_altitude_measurement(time,alongshore), float64 min_altitude_measurement(time,alongshore), float64 time_topo(time,alongshore), float64 time_bathy(time,alongshore), int16 origin(time,alongshore,cross_shore), float64 altitude(time,alongshore,cross_shore)
    groups:

>>>
```

It must be an issue with your netcdf-c installation.  Or, maybe you have multiple netcdf-c installations on your system, and when you build netcdf4-python it is linking the wrong one.

**Comment by jswhit** [comment]
BTW - you can use setup.cfg to control the libraries and headers that netcdf4-python uses.

**Comment by evanmason** [comment]
Thanks for the reply.  So, I reinstalled the netcdf C lib, changing from version 4.4.0. to 4.4.1.

```
[emason@marula Downloads]$ nc-config --all

This netCDF 4.4.1 has been built with the following features: 

  --cc        -> gcc
  --cflags    ->  -I/usr/local/include -I/opt/netcdf_hdf5/hdf5-1.8.16/hdf5/include
  --libs      -> 

  --has-c++   -> no
  --cxx       -> 
  --has-c++4  -> no
  --cxx4      -> 

  --fc        -> gfortran
  --fflags    -> -I/usr/local/include
  --flibs     -> -L/usr/local/lib -lnetcdff -L/usr/local/lib -lnetcdf -lnetcdf
  --has-f90   -> no
  --has-f03   -> yes

  --has-dap   -> yes
  --has-nc2   -> yes
  --has-nc4   -> yes
  --has-hdf5  -> yes
  --has-hdf4  -> no
  --has-logging-> no
  --has-pnetcdf-> no
  --has-szlib -> 

  --prefix    -> /usr/local
  --includedir-> /usr/local/include
  --version   -> netCDF 4.4.1
```

I then reinstalled netCDF4, noting during the build that it is linking correctly to 4.4.1
```
using nc-config ...
using netcdf library version 4.4.1
```
Unfortunately, as before, using ncdump I can read the url, but not with netCDF4 which still gives the error `IOError: NetCDF: Unknown file format`.  I also tried on a colleague's machine (identical OS), that I set up a few months ago, and there it does work.  This is puzzling.

**Comment by evanmason** [comment]
Further to my last post, I notice that:
```
In [6]: import netCDF4

In [7]: netCDF4.__netcdf4libversion__
Out[7]: u'4.4.0'
```
so it looks like an old netCDF4 is somehow being called, but if so I'm not sure how to find and eliminate it:
```
[emason@marula pytools]$ locate _netCDF4.so
/home/emason/.python-eggs/netCDF4-1.2.4-py2.7-linux-x86_64.egg-tmp/netCDF4/_netCDF4.so
/opt/python27/netCDF4-1.2.4/build/lib.linux-x86_64-2.7/netCDF4/_netCDF4.so
[emason@marula pytools]$ 
[emason@marula pytools]$ 
[emason@marula pytools]$ ls -l /home/emason/.python-eggs/netCDF4-1.2.4-py2.7-linux-x86_64.egg-tmp/netCDF4/_netCDF4.so 
-rwxr-xr-x 1 emason emason 4839705 Sep  1 18:07 /home/emason/.python-eggs/netCDF4-1.2.4-py2.7-linux-x86_64.egg-tmp/netCDF4/_netCDF4.so*
[emason@marula pytools]$ ls -l /opt/python27/netCDF4-1.2.4/build/lib.linux-x86_64-2.7/netCDF4/_netCDF4.so
-rwxr-xr-x 1 root root 4839705 Sep  1 18:07 /opt/python27/netCDF4-1.2.4/build/lib.linux-x86_64-2.7/netCDF4/_netCDF4.so*
[emason@marula pytools]$
```
These two have the same timestamp that corresponds to the installation linked to 4.4.1.

**Comment by evanmason** [comment]
Ok, solved.  I installed netCDF4 v1.2.5 from github, and was able to verify the upgrade.  During installation I still got the line `using netcdf library version 4.4.1` but upon checking after installation found `netCDF4.__netcdf4libversion__` returning 4.4.0.

I did have an old 4.4.0 installation on my system, so I made that into a hidden file, and recompiled netCDF4 v1.2.5.  Opendap is now working.  I now need to check my bashrc and clean up my paths that I suspect caused this confusion.
